applications/vanilla/settings/class.hooks.php

```diff
@@ -1110,6 +1110,23 @@ public function pluginController_tagsearch_create() {
         redirectTo(url('/tags/search'.($query ? '?'.$query : null)), 301);
     }
 
+    /**
+     * Hook in before a discussion is rendered and display any messages.
+     *
+     * @param mixed DiscussionController $sender
+     * @param array array $args
+     */
+    public function discussionController_beforeDiscussionDisplay_handler($sender, array $args) {
+        if (!is_object($sender) || !method_exists($sender, "getMessages")) {
```

Comment:
Any reason we're not just doing an `instanceof` check here?

Reply:
Being overly defensive. A `instanceof` check will probably be sufficient.